crates/ingress-http/src/handler/awakeables.rs

```diff
@@ -98,8 +98,7 @@ where
             );
             self.dispatcher
                 .send_invocation_response(InvocationResponse {
-                    id: invocation_id,
-                    entry_index,
+                    target: JournalCompletionTarget::for_v3_completions(invocation_id, entry_index),
```

Comment:
I think we should capture somewhere why Awakeables are not fenced off by an invocation epoch. Would be great to capture a few more details and thoughts wrt to the signals.

Reply:
I wrote this already here https://github.com/restatedev/restate/issues/2890#issuecomment-2804150578

Reply:
>  If I understand things correctly, then awakeables will always get the invocation epoch 0. If I now time travelled to a point where the awakeable hasn't been sent (trim point being before the awakeable command), future completions should only be accepted if they have now an invocation_epoch + 1.

You misunderstood this. `ExternalSignalIdentifier` is for awakeables of protocol >= v4, `AwakeableIdentifier` is for awakeables of protocol <= 3. You cannot trim a journal of protocol <= 3, so assuming always epoch 0 is fine.

Reply:
I added a comment.

Reply:
> I wrote this already here https://github.com/restatedev/restate/issues/2890#issuecomment-2804150578

I think it would be nice to capture this a little bit closer to the code to make it more discoverable. I fear that this Github comment will probably be missed in the future quite easily.